ConfigSpace/hyperparameters.pyx

```diff
@@ -806,23 +843,22 @@ cdef class UniformIntegerHyperparameter(IntegerHyperparameter):
         value = self._inverse_transform(value)
         return value
 
-    def _transform(self, vector: Union[np.ndarray, float, int]) -> Optional[Union[np.ndarray, float, int]]:
+    cpdef np.ndarray _transform_vector(self, np.ndarray vector):
+        if np.any(~np.isfinite(vector)):
+            raise ValueError()
+        vector = self.ufhp._transform_vector(vector)
+        if self.q is not None:
+            vector = np.rint(vector / self.q) * self.q
 
-        # Avoid expensive call to np.any() if not necessary
-        if isinstance(vector, np.ndarray):
-            if np.any(np.isnan(vector)):
-                return None
-        elif np.isnan(vector):
-            return None
+        return np.rint(vector)
 
-        vector = self.ufhp._transform(vector)
+    cpdef long _transform_scalar(self, double scalar):
+        if not math.isfinite(scalar):
```

Comment:
I'm only seeing this now, but you replaced the check for NaN with a check for being finite. I don't think this is necessary as there are usually no infs/-infs around. Also, it's much faster to only check for NaN:
```
In [13]: %timeit math.isfinite(np.NaN)                                                                
127 ns ± 1.93 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [14]: %timeit np.NaN == np.NaN                                                                     
89.8 ns ± 0.833 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
Okay sounds good!